wagtail/embeds/rich_text.py

```diff
@@ -3,55 +3,50 @@
 from wagtail.admin.rich_text.converters import editor_html
 from wagtail.admin.rich_text.converters.contentstate_models import Entity
 from wagtail.admin.rich_text.converters.html_to_contentstate import AtomicBlockEntityElementHandler
+from wagtail.core.rich_text.feature_registry import LinkHandler
 from wagtail.embeds import embeds, format
+from wagtail.embeds.embeds import get_embed
 from wagtail.embeds.exceptions import EmbedException
+from wagtail.embeds.models import Embed
 
 
-# Front-end conversion
+# Front-end + hallo.js / editor-html conversions
 
-def media_embedtype_handler(attrs):
-    """
-    Given a dict of attributes from the <embed> tag, return the real HTML
-    representation for use on the front-end.
-    """
-    return format.embed_to_frontend_html(attrs['url'])
+class EmbedHandler(LinkHandler):
+    link_type = 'media'
 
+    @staticmethod
+    def get_model():
+        return Embed
```

Comment:
I don't think it's logically correct to treat EmbedHandler as being associated with the `Embed` model - the `Embed` model is more like a cache, and deleting records from that table has no consequence on rich text fields that reference that URL. (Looking at it more abstractly: there's no reason that a linktype/embedtype _has_ to be associated with a model; it's just rewriting one kind of HTML tag into another.)

Happily, I don't think there's any negative consequence from simply returning `None` here instead.

Reply:
I made this for more than just deleting. The idea is to have a way, through collectors, to find all references of a single object. This gives a lot of possibilities, including the possibility to create a task to clean the embeds from time to time. If a reference to an embed is removed, the object still sits there in the database, useless. On huge sites, that’ll be good to have a command to remove useless embeds from time to time, hence the system to find references for it in rich text & stream fields.

Indeed, linktype/embedtypes can do something without a model, but this is not the case in all the linktype/embedtypes provided by Wagtail.

Reply:
If you're using it for garbage-collecting the embeds table, then calling `get_embed` from `get_instance` will do exactly the wrong thing... for any embed not found in the table, it will make a round-trip to the embed provider and create a new record.

Reply:
OK, fair point! On a related note, I think we could optimise things further by having the Link/EmbedHandler classes implement a method to check "do these tag attributes correspond to one of these model instances", so that we're not uselessly instantiating every object we encounter in rich text (only to either throw it away or match it to an object we already have). But at that point I'm gift-horsing ;-)